Upgrade salt to 3002.5

TeddyAndrieux opened the pull request:
**Component**:

'salt'

**Context**: 

Some CVE was discovered in Salt 3002.2 https://saltproject.io/security_announcements/active-saltstack-cve-release-2021-feb-25/

**Summary**:

- Bump Salt version to 3002.2
- Add `roster` in Salt master configuration file (sees https://github.com/saltstack/salt/issues/59657)

---

TeddyAndrieux committed:
buid,tests: Bump Salt to `3002.5`

TeddyAndrieux committed:
salt: Set roster to `kubernetes` directly in Salt master config

Add roster directly in salt master config so that it's simpler to do
some manual call to `salt-ssh` directly.

Also in Salt 3002.5 `roster` from kwargs in ssh command is ignored,
sees https://github.com/saltstack/salt/issues/59657

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/upgrade-salt-30025` with contents from `improvement/upgrade-salt-30025`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/upgrade-salt-30025 origin/development/2.8
 $ git merge origin/improvement/upgrade-salt-30025
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/upgrade-salt-30025
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.9/improvement/upgrade-salt-30025` with contents from `w/2.8/improvement/upgrade-salt-30025`
and `development/2.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.9/improvement/upgrade-salt-30025 origin/development/2.9
 $ git merge origin/w/2.8/improvement/upgrade-salt-30025
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.9/improvement/upgrade-salt-30025
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

alexandre-allard commented on images/salt-master/Dockerfile:
```diff
@@ -3,7 +3,7 @@ FROM centos:7.6.1810
 
 # Versions to use
-ARG SALT_VERSION=3002.2
+ARG SALT_VERSION=3002.5
```
The Salt version is already passed to the container build context from the buildchain, it should be:
```suggestion
ARG SALT_VERSION
```
This way we don't have to maintain the version in 2 places.

TeddyAndrieux replied:
Right, it's not linked to this PR, but I agree

TeddyAndrieux replied:
Done

TeddyAndrieux committed:
images: Do not longer put a default value for SALT_VERSION in Dockerfile

Do not longer provide the SALT_VERSION default value in the Dockerfile,
this docker file is supposed to be used with the right Salt version
directly in the argument
```
docker build . --build-arg SALT_VERSION=<salt_version>
```

In the buildchain we override this value automatically with the one in
versions.py, so in case of Salt upgrade the one in the dockerfile do not
need to be maintained

bert-e commented:
# History mismatch



Merge commit #75d22962bfeb70a9f7df6e62576331ca46015f50 on the integration branch
`w/2.8/improvement/upgrade-salt-30025` is merging a branch which is neither the current
branch `improvement/upgrade-salt-30025` nor the development branch
`development/2.8`.

It is likely due to a rebase of the branch `improvement/upgrade-salt-30025` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.8/improvement/upgrade-salt-30025` with contents from `improvement/upgrade-salt-30025`
and `development/2.8`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.8/improvement/upgrade-salt-30025 origin/development/2.8
 $ git merge origin/improvement/upgrade-salt-30025
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.8/improvement/upgrade-salt-30025
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

TeddyAndrieux commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.7`

* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
